lib/active_storage/service/open_stack_service.rb

```diff
@@ -143,6 +143,10 @@ def update_metadata(key, content_type:, disposition: nil, filename: nil, **)
 
   private
 
+    def head_for(key, &block)
+      client.head_object(container, key, &block)
```

Comment:
[`#head_object`](https://github.com/fog/fog-openstack/blob/master/lib/fog/openstack/storage/requests/head_object.rb)  does not accept a block.

```ruby
 def head_for(key)
      client.head_object(container, key)
end
```